src/refmterr/lib/reporter.re

```diff
@@ -230,85 +232,128 @@ module Make = (Styl: Stylish.StylishSig) => {
+         line right after our supposedly parsed and pretty-printed error to make them
+         look like one printed error. */
+      /* the effing length we'd go for better errors... someone gimme a cookie */
+      | ErrorFile(NonexistentFile) =>
+        /* this case is never reached because we don't ever return `ErrorFile NonexistentFile` from
+           `ParseError.specialParserThatChecksWhetherFileEvenExists` */
+        originalRevLines
+      | ErrorFile(Stdin(original)) => [
           sp(
-            "%s Cannot find file %s.",
+            "%s (from stdin - see message above)",
             red(~bold=true, "Error:"),
-            red(~bold=true, ~underline=true, filename),
           ),
-          ...originalRevLines,
-        ];
-      } else {
-        [
+          original,
+        ]
+      | ErrorFile(CommandLine(moduleName)) => [
+          "",
           sp(
-            "%s Cannot find file %s.",
+            "%s module %s not found.",
             red(~bold=true, "Error:"),
-            red(~bold=true, filename),
+            red(~underline=true, ~bold=true, moduleName),
           ),
           ...originalRevLines,
+        ]
+      | ErrorFile(NoneFile(filename)) =>
+        /* TODO: test case for this. Forgot how to repro it */
+        if (Filename.check_suffix(filename, ".cmo")) {
+          [
+            "Cmo files are artifacts the compiler looks for when compiling/linking dependent files.",
+            sp(
+              "%s Cannot find file %s.",
+              red(~bold=true, "Error:"),
+              red(~bold=true, ~underline=true, filename),
+            ),
+            ...originalRevLines,
+          ];
+        } else {
+          [
+            sp(
+              "%s Cannot find file %s.",
+              red(~bold=true, "Error:"),
+              red(~bold=true, filename),
+            ),
+            ...originalRevLines,
+          ];
+        }
+      | ErrorContent(withFileInfo) =>
+        List.concat([
+          ["", ""],
+          ReportError.report(~refmttypePath, withFileInfo.parsedContent),
+          [""],
+          printFile(withFileInfo),
+          [""],
+          [""],
+          indent(dim("# "), List.map(dim, originalRevLines)),
+          [highlight(~dim=true, ~bold=true, "# Unformatted Error Output:")],
+        ])
+      | Warning(withFileInfo) =>
+        List.concat([
+          ["", ""],
+          ReportWarning.report(
+            ~refmttypePath,
+            withFileInfo.parsedContent.code,
+            withFileInfo.filePath,
+            withFileInfo.parsedContent.warningType,
+          ),
+          [""],
+          printFile(
+            ~isWarningWithCode=withFileInfo.parsedContent.code,
+            withFileInfo,
+          ),
+          [""],
+          [""],
+          indent(dim("# "), List.map(dim, originalRevLines)),
+          [
+            highlight(~dim=true, ~bold=true, "# Unformatted Warning Output:"),
+          ],
+        ])
+      };
+    rawOutput ?
```

Comment:
can we extract a variable or a function for the first part of the ternary? I was confused before and only now realized what is happening. Other than that, looks good